pkg/controller/mutators/assignmetadata/assignmetadata_controller.go

```diff
@@ -195,6 +208,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
+	// Since we aren't deleting the mutator, we are either adding or updating
+	ingestionStatus := ctrlmutators.MutatorStatusError
+	defer func() {
```

Comment:
This deferred function is complex enough that it warrants its own method.

Reply:
done